File: src/api/AMP.EMS.API/Core/Entities/Event.cs
using AMP.Infrastructure.Entity;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.EntityFrameworkCore;
using Newtonsoft.Json;

namespace AMP.EMS.API.Core.Entities;

public class Event : FullAuditableEntity<Guid>
{
    public required string Title { get; set; }

    public required Guid EventTypeId { get; set; }

    public string Description { get; set; } = string.Empty;

    public string Location { get; set; }

    public int MaxGuests { get; set; }

    public required DateTime StartDate { get; set; }

    public required DateTime EndDate { get; set; }
}

File: src/ui/AMP.EMS/src/app/core/models/event.ts
import { EventGuestRole } from "./event-guest-role";
import { EventType } from "./event-type";

export interface Event {
    id?: string;
    title?: string;
    eventTypeId?: string;
    eventType?: EventType;
    startDate?: Date;
    endDate?: Date;
    eventRoles?: EventGuestRole[];
    description?: string;
}